api/api-pipelines-as-code-misc-v1/src/main/java/com/thoughtworks/go/apiv1/pipelinesascodemisc/PipelinesAsCodeMiscControllerV1.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright 2019 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.apiv1.pipelinesascodemisc;
+
+import com.thoughtworks.go.api.ApiController;
+import com.thoughtworks.go.api.ApiVersion;
+import com.thoughtworks.go.api.base.OutputWriter;
+import com.thoughtworks.go.api.representers.JsonReader;
+import com.thoughtworks.go.api.spring.ApiAuthenticationHelper;
+import com.thoughtworks.go.api.util.GsonTransformer;
+import com.thoughtworks.go.api.util.HaltApiResponses;
+import com.thoughtworks.go.api.util.MessageJson;
+import com.thoughtworks.go.apiv8.admin.shared.representers.PipelineConfigRepresenter;
+import com.thoughtworks.go.apiv8.admin.shared.representers.stages.ConfigHelperOptions;
+import com.thoughtworks.go.config.ConfigRepoPlugin;
+import com.thoughtworks.go.config.CruiseConfig;
+import com.thoughtworks.go.config.GoConfigPluginService;
+import com.thoughtworks.go.config.PipelineConfig;
+import com.thoughtworks.go.config.exceptions.HttpException;
+import com.thoughtworks.go.config.materials.PasswordDeserializer;
+import com.thoughtworks.go.config.update.CreatePipelineConfigCommand;
+import com.thoughtworks.go.plugin.access.configrepo.ExportedConfig;
+import com.thoughtworks.go.server.service.GoConfigService;
+import com.thoughtworks.go.server.service.PipelineConfigService;
+import com.thoughtworks.go.spark.spring.SparkSpringController;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import spark.HaltException;
+import spark.Request;
+import spark.Response;
+
+import java.util.function.Consumer;
+
+import static com.thoughtworks.go.api.util.HaltApiResponses.haltBecauseOfReason;
+import static com.thoughtworks.go.spark.Routes.PaC.BASE_INTERNAL_API;
+import static com.thoughtworks.go.spark.Routes.PaC.PREVIEW;
+import static java.lang.String.format;
+import static spark.Spark.*;
+
+@Component
+public class PipelinesAsCodeMiscControllerV1 extends ApiController implements SparkSpringController {
```

Comment:
I don't love naming a class miscellaneous, but it's a nitpick. I feel like `PipelinesAsCodeInternalControllerV1` would be fine.

It's 100% up to you. Changing the name or not doesn't affect the merge-ability of this PR.

Reply:
Yeah, that's probably better to be explicit that this is not public.